### ISSUE TITLE
Needed transaction log backups can be deleted

### ISSUE THREAD

**Issue by rwhoward** [olahallengren/sql-server-maintenance-solution#20]
About a week ago we had a system that could not be fully restored. We had a full backup from roughly 48 hours before we attempted the restore. The CleanupTime parameter for the transaction log backups was set to 24.

In the script it looks like the database_backup_lsn is being used to determine the last valid transaction log which can be cleaned up. I'm not sure that's the right value to check though. 

Based on what I've been able to read on the LSN (I am by no means an expert), the full backup's last_lsn should fall between the first log's first_lsn and last_lsn. Or... log.first_lsn <= full.last_lsn &&log.last_lsn > full.last_lsn. 

In our case that log file was about 10 files too early and was deleted out. That lines up with our problem scenario - SQL errored out saying the file we were trying to restore was too new. I've got the output of the backup metadata if that would be helpful but unfortunately I don't have the output of the errors. 

My guess is this isn't normally an issue. This server has an exceptionally long full backup time and it may have thrown off the order from the expected. But if it can happen here then it might happen somewhere else too. And if you need to recover to full + logs instead of full + diff + logs that's a problem.

tl;dr
Change the CleanupTime safety check to use last_lsn instead of database_backup_lsn.

**Comment by olahallengren** [comment]
I am sorry for not getting back on this before. I will review this code and get back.

**Comment by olahallengren** [comment]
Do you have availability groups on this server?

Are all backups running on the primary?

**Comment by olahallengren** [comment]
I am sorry for not getting to this issue before, but it has been fixed now.
